tests/integration/test_dictionaries_update_and_reload/test.py

```diff
@@ -250,6 +260,15 @@ def test_reload_after_fail_by_timer(started_cluster):
     assert expected_error in instance.query_and_get_error(
         "SELECT dictGetInt32('no_file_2', 'a', toUInt64(9))"
     )
+
+    # on sanitizers builds it can return 'FAILED_AND_RELOADING' which is not quite right
+    # add retry for these builds
+    if (
+        instance.is_built_with_sanitizer()
+        and get_status("no_file_2") == "FAILED_AND_RELOADING"
```

Comment:
Why it happens?

And it still fails - https://s3.amazonaws.com/clickhouse-test-reports/66311/316dd790bdb5a8cb4aef11c356f14700f3b9169e/integration_tests__asan__old_analyzer__[3_6].html

Reply:
Anyway this code is wrong, since you still can get `FAILED_AND_RELOADING` two lines below, there should be `else` at least.

Reply:
- https://github.com/ClickHouse/ClickHouse/pull/67754

Reply:
https://github.com/ClickHouse/ClickHouse/pull/67754#issuecomment-2267598185